rq/worker.py

```diff
@@ -668,20 +681,39 @@ def unsubscribe(self):
             self.pubsub.unsubscribe()
             self.pubsub.close()
 
-    def reorder_queues(self, reference_queue):
-        """Method placeholder to workers that implement some reordering strategy.
-        `pass` here means that the queue will remain with the same job order.
+    def reorder_queues(self, reference_queue: 'Queue'):
+        """Reorder the queues according to the strategy.
+        As this can be defined both in the `Worker` initialization or in the `work` method,
+        it doesn't take the strategy directly, but rather uses the private `_dequeue_strategy` attribute.
 
         Args:
-            reference_queue (Union[Queue, str]): The queue
+            reference_queue (Union[Queue, str]): The queues to reorder
         """
-        pass
+        if self._dequeue_strategy is None:
+            self._dequeue_strategy = DequeueStrategy.DEFAULT
+
+        if self._dequeue_strategy not in ("default", "random", "round_robin"):
+            self.log.warning(
```

Comment:
Hey if we're checking this, I think we should just raise an exception. Mind adding this scenario to tests also?

Reply:
Not sure this scenario is critical enough to interrupt execution, I think logging at `warning` level is enough in this case.

Reply:
If there's a check here and the value is not expected, we should raise an exception :).

`Errors should never pass silently.` - Zen of Python ;)

Reply:
At this point we would be discussing whether a `warning` is "silent", but let's move forward on this - which kind of exception would you prefer here?

Reply:
Just a regular `ValueError` would do.